include/soci/callbacks.h

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright (C) 2015 Maciej Sobczak
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef SOCI_CALLBACKS_H_INCLUDED
+#define SOCI_CALLBACKS_H_INCLUDED
+
+namespace soci
+{
+
+class session;
+
+// Simple callback interface for reporting failover events.
+// The meaning of each operation is intended to be portable,
+// but the behaviour details and parameters can be backend-specific.
+class SOCI_DECL failover_callback
+{
+public:
+
+    // Called when the failover operation has started,
+    // after discovering connectivity problems.
+    virtual void started() {}
```

Comment:
No. The failover_callback is a classical interface, which does not enforce any policy with regard to exception handling. It is the choice of the backend whether the calls to this interface should be protected by try/catch block or not and I can imagine a backend where this is not necessary. I can also imagine a user who decides to use the same interface to implement some forwarding or notification mechanism at the application level, and it is also not the responsibility of the interface to enforce such policies.